src/wing/TiglWingHelperFunctions.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2019 CFS Engineering
+ *
+ * Created: 2019 Malo Drougard <malo.drougard@protonmail.com>
+ * Author: Malo Drougard
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "TiglWingHelperFunctions.h"
+
+#include "tigl.h"
+#include "CCPACSWingSegment.h"
+
+namespace tigl
+{
+
+namespace winghelper
+{
+
+bool HasShape(const tigl::CCPACSWing& wing)
+{
+    return wing.GetSegmentCount() > 0;
+
+}
+
+TiglAxis GetDeepDirection(const tigl::CCPACSWing& wing)
+{
+    if (!HasShape(wing)) {
+        LOG(WARNING) << "CTiglWingHelper::GetDeepDirection: This wing has no shape -> impossible to determine the "
+                        "direction properly. The default direction will be returned";
+        return TIGL_X_AXIS;
+    }
+
+    gp_XYZ cumulatedDepthDirection(0, 0, 0);
+    for (int i = 1; i <= wing.GetSegmentCount(); ++i) {
+        const CCPACSWingSegment& segment = wing.GetSegment(i);
+        gp_XYZ dirDepth                  = segment.GetChordPoint(0, 1).XYZ() - segment.GetChordPoint(0, 0).XYZ();
+        dirDepth                         = gp_XYZ(fabs(dirDepth.X()), fabs(dirDepth.Y()), fabs(dirDepth.Z()));
+        cumulatedDepthDirection += dirDepth;
+    }
+    const CCPACSWingSegment& outerSegment = wing.GetSegment(wing.GetSegmentCount());
+    gp_XYZ dirDepth = outerSegment.GetChordPoint(1, 1).XYZ() - outerSegment.GetChordPoint(1, 0).XYZ();
+    dirDepth        = gp_XYZ(fabs(dirDepth.X()), fabs(dirDepth.Y()), fabs(dirDepth.Z()));
+    cumulatedDepthDirection += dirDepth;
+
+    switch (GetMajorDirection(wing)) {
```

Comment:
If I understand the function correctly, in `GetMajorDirection` first the heuristic is used to dermine the depth direction, and then from that the major direction is calculated and the depth direction is disregarded. Then here, we calculate the depth direction again using the same heuristic (DRY), but dependent on the outcome of `GetMajorDirection`?

Reply:
I actually don't know ;) (see comment below). Good point though